llama-index-core/llama_index/core/workflow/utils.py

```diff
@@ -54,12 +55,26 @@ def validate_step_signature(fn: Callable) -> None:
         raise WorkflowValidationError(msg)
 
 
-def get_steps_from_class(_class: object) -> dict:
+def get_steps_from_class(_class: object) -> Dict[str, Callable]:
     """Given a class, return the list of its methods that were defined as steps."""
     step_methods = {}
-    all_methods = inspect.getmembers(_class, predicate=inspect.ismethod)
+    all_methods = inspect.getmembers(_class, predicate=inspect.isfunction)
 
     for name, method in all_methods:
+        print(name, hasattr(method, "__step_config"))
+        if hasattr(method, "__step_config"):
+            step_methods[name] = method
+
+    return step_methods
+
+
+def get_steps_from_instance(workflow: object) -> Dict[str, Callable]:
```

Comment:
This is the old function used to create the asyncio tasks: the returned functions are actual methods of the current workflow instance. I changed the name to reflect the fact that it expects an instance.